### PR TITLE
Selective Linting

### PR DESCRIPTION
**Pull request by mokagio**
This PR implements @thiagofelix suggestion from #16 to re-enable selective linting.

I'd like for the functionality to be as frictionless as possible. To do so I opted to generate a config YML on the fly based on the give one (if any) but with the `included` configuration remove, so that the `--path` option will take place.

I haven't found a clean way to do this yet, the code is a bit convoluted at the moment.

~~I also had a blast at implementing inline warnings, because I wanted them for my team, but this might be out of scope from the PR.~~ _Will make a dedicated PR for that._

### CODE REVIEW

File: lib/danger_plugin.rb [hunk 1]
```diff
@@ -39,33 +39,67 @@ def lint_files
...
-      errors = result_json.select do |results| 
-        results['severity'] == 'Error' 
-      end
-
-      message = ''
-
-      # We got some error reports back from swiftlint
-      if warnings.count > 0 || errors.count > 0
-        message = "### SwiftLint found issues\n\n"
+      require 'tempfile'
```
Comment (mokagio) [inline lib/danger_plugin.rb]: I wouldn't usually put `require` inline, but that's what I've seen for the `json` required down in this method.

File: lib/danger_plugin.rb [hunk 2]
```diff
@@ -39,33 +39,67 @@ def lint_files
...
+      Tempfile.open('.swiftlint_danger.yml') do |temp_config_file|
```
Comment (mokagio) [inline lib/danger_plugin.rb]: Ensuring that temporary files are properly cleaned up is important, using this block based syntax seems like a good way to ensure it's always cleaned up. The disadvantage is that it adds a bit of complexity to the method.

File: lib/danger_plugin.rb [hunk 3]
```diff
@@ -39,33 +39,67 @@ def lint_files
...
+
+        # We got some error reports back from swiftlint
+        if warnings.count > 0 || errors.count > 0
+          message = "### SwiftLint found issues\n\n"
+        end
+
+        warnings.each do |warning|
+          location = warning['file'].gsub(Dir.pwd, '')
+          warn(warning['type'], file: location, line: warning['line'])
+        end
```
Comment (mokagio) [inline lib/danger_plugin.rb]: As mentioned in the PR description, this was an experiment, might be better to have a dedicated PR for this.

**Comment by mokagio** [comment]
I also do realise that it might have been simpler to enhance SwiftLint, but fixing this seemed like a quicker first step  ¯\_(ツ)_/¯.

**Comment by mokagio** [comment]
Hey @ashfurrow, thanks for having a look 😄 .

I've pushed a new commit with a little fix for an edge case.

I think this is ready to merge yeah.

Cheers 🍻 .

**Comment by valeriomazzeo** [comment]
@ashfurrow I think this is now ignoring the original config file `excluded`?

**Comment by freak4pc** [comment]
@valeriomazzeo I'm experiencing the same issue, It's ignoring my excludes statement in `.swiftlint.yml` 🤔
